src/metabase/server/middleware/security.clj

```diff
@@ -143,14 +158,16 @@
 
 (defn- add-security-headers* [request response]
   (update response :headers merge (security-headers
+                                   :nonce          (:nonce request)
                                    :allow-iframes? ((some-fn request.u/public? request.u/embed?) request)
                                    :allow-cache?   (request.u/cacheable? request))))
 
 (defn add-security-headers
-  "Add HTTP security and cache-busting headers."
+  "Middleware that adds HTTP security and cache-busting headers."
   [handler]
   (fn [request respond raise]
-    (handler
-     request
-     (comp respond (partial add-security-headers* request))
-     raise)))
+    (let [request (assoc request :nonce (generate-nonce))]
```

Comment:
@metabase/backend-developers Note this is redundant because we only need the nonce for rendering the `index.html`, `public.html`, and `embed.html`, yet we are generating one for every request with security headers. I think it is simpler to do it for every request rather than test for specific paths here, and generating the nonce takes negligible time anyway.

From here, the request map carries the `nonce` value to both the `security-headers` function in this namespace, and `metabase.server.routes.index/load-entrypoint-template` below.